### ISSUE TITLE
ps_accounts will not link my store to my account

### ISSUE THREAD

**Issue by LarsGMortensen** [PrestaShop/PrestaShop#34038]
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md).
- [X] I have already [searched in existing issues](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3ABug) and found no previous report of this bug.

### Describe the bug and add attachments

On PrestaShop 8.1.1
with PrestaShop Account v6.2.6
and PrestaShop Marketplace in your Back Office v4.5.3
and PrestaShop Marketing with Google v1.59.0

In order to start using PrestaShop Marketing with Google, it says that I must link my store to my account.
But when I try to do that, it responds with this error: "1 shop could not be linked. The shop has not been verified. Reason : Error while getting domain"
Please see the screen-recording (below) from start to end for visual explanation.

I suspect this has something to do with the fact that my PrestaShop is installed on a subdomain, however that can't be changed. I need to have PrestaShop running on a subdomain, and quite frankly it seems less than optimal that this has not been factored in when creating ps_accounts.

![1h5Z6K7kV1](https://github.com/PrestaShop/PrestaShop/assets/26773745/b7f10257-ee71-4992-84b0-a8870c7486f5)

### Expected behavior

I would expect to be able to link my store to my account with ps_accounts. Especially as ps_accounts is an official module, developed by Prestashop.

### Steps to reproduce

1. Install PrestaShop 8.1.1 on a subdomain! (in my case it is installed on prestashop8.example.com)
with PrestaShop Account v6.2.6
and PrestaShop Marketplace in your Back Office v4.5.3
and PrestaShop Marketing with Google v1.59.0

2. Try to configure PrestaShop Marketing with Google, and link the store to ps_accounts.

Please see this screen-recording from start to end for visual explanation:

![1h5Z6K7kV1](https://github.com/PrestaShop/PrestaShop/assets/26773745/edd068b9-5249-4c7d-8c3d-f205cf26dff3)

### PrestaShop version(s) where the bug happened

8.1.1

### PHP version(s) where the bug happened

7.4.33

### If your bug is related to a module, specify its name and its version

ps_accounts v6.2.6, psxmarketingwithgoogle v1.59.0

**Comment by MatShir** [comment]
I'm sharing this issue to the team thanks for the report. But since the issue is not part of the open source project I'm closing this issue. 
If you want to contact again the developers in charge of the modules, please fulfill this [form](https://help-center.prestashop.com/en/contact).
Thanks in advance.

**Comment by antoinemetifeu** [comment]
Hi @LarsGMortensen your shop/server answer with an HTTP Code 555 with this HTML response:

```
<html>
  <head>
   <title>555 Security Incident Detected</title>
    <meta name='viewport' content='width=device-width'>
    <link href='//splash.simply.com/errorpage.css' rel='stylesheet'>
  </head>
<body>
<h1>555 Security Incident Detected</h1>
<p>Your request was blocked.</p>

<h3>If you are the owner of the website:</h3>
<p>The Website Application Firewall that is protecting your website has blocked this request for being suspicious.</p>
<p>You can see the detailed reason for this in your webserver logs.</p>

</body>
</html>
```

Our API is blocked by our shop server

**Comment by LarsGMortensen** [comment]
Hi @antoinemetifeu 
Thank you for your update on the matter. I have now presented your answer for the technicians who runs our hosting. They reply that their firewall will block all requests that:

- Are missing a user agent
- Uses very generic user agents (i.e. python-requests og curl)
- Uses obviously fake user-agents

They also told me that if you supply your own user-agent on these requests, then they won’t be blocked. There are no other requirements, so a user agent like “My user-agent” will be allowed access without any problems.

Is there a way that I can override the script that makes the call to the server, so that I can add a user-agent and make it work?

Alternatively they can also whitelist specific IP-addresses, so if you can let me know which IP the request comes from, then I can make them whitelist that IP?

**Comment by antoinemetifeu** [comment]
Hello @LarsGMortensen

The next week, we will add a custom user agent on our all outgoing requests after the PR pass all the tests.

Hope this resolves your issue.

Thanks !
